lib/metadataTypes/MetadataType.js

```diff
@@ -311,6 +311,59 @@ class MetadataType {
         return;
     }
 
+    /**
+     * test if metadata was actually changed or not to potentially skip it during deployment
+     *
+     * @param {TYPE.MetadataTypeItem} cachedVersion cached version from the server
+     * @param {TYPE.MetadataTypeItem} metadata item to upload
+     * @param {string} [fieldName] optional field name to use for identifying the record in logs
+     * @returns {boolean} true if metadata was changed
+     */
+    static hasChanged(cachedVersion, metadata, fieldName) {
```

Comment:
should we not just use hasChangedGeneric and if it is overriden then the override applies?

Reply:
ideally yes. but we have too many cases where whats in cache doesn't necessarily match what we deploy. therefore, i'd rather opt-in per type than opt-out.